### PR TITLE
support false-positive fast math

### PR DESCRIPTION
**Pull request by cery999**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

**Comment by cery999** [comment]
@alexeyr hi, fix the false positive for fast math
case is 
nan < 9 ? nan : 9
1 < nan ? 1 : nan
which will case nan to be number

**Comment by wzh99** [comment]
@cery999 I am the author of #11696. Thanks for your efforts in sending this PR. The reason why I did not fix this myself is that I have no idea whether this inconsistency should be regarded as a bug indeed, or an undefined behavior. 

@masahi Would you please take a look at this PR and decide whether we should fix the inconsistency between `tanh` and `fast_tanh` on `nan` input, or leave it alone?

**Review by masahi — APPROVED** [review]
The changes makes sense to me.

**Comment by masahi** [comment]
@tvm-bot rerun

**Comment by wzh99** [comment]
@masahi The Android CI still fails. It seems that there is something wrong with the Android CI environment. There is an error in "Build android_camera":
```
Traceback (most recent call last):
  File "prepare_model.py", line 139, in <module>
    main(model, output_path)
  File "prepare_model.py", line 94, in main
    net, params = get_model(model_str)
  File "prepare_model.py", line 63, in get_model
    import keras
  File "/home/runner/.local/lib/python3.8/site-packages/keras/__init__.py", line 21, in <module>
    from keras import models
  File "/home/runner/.local/lib/python3.8/site-packages/keras/models/__init__.py", line 18, in <module>
    from keras.engine.functional import Functional
  File "/home/runner/.local/lib/python3.8/site-packages/keras/engine/functional.py", line 27, in <module>
    from keras.dtensor import layout_map as layout_map_lib
  File "/home/runner/.local/lib/python3.8/site-packages/keras/dtensor/layout_map.py", line 25, in <module>
    from keras.dtensor import lazy_variable
  File "/home/runner/.local/lib/python3.8/site-packages/keras/dtensor/lazy_variable.py", line 26, in <module>
    from tensorflow.python.trackable import base as trackable
ModuleNotFoundError: No module named 'tensorflow.python.trackable'
Error: Process completed with exit code 1.
```
I also notice that there is another error message in CI: 
```
ERROR: tensorflow 2.9.2 has requirement keras<2.10.0,>=2.9.0rc0, but you'll have keras 2.10.0 which is incompatible.
```
I guess the failure is caused by the incompatible version of Keras with TensorFlow.

**Comment by cery999** [comment]
> @tvm-bot rerun

from tensorflow.python.trackable import base as trackable
[150](https://github.com/apache/tvm/runs/8194885989?check_suite_focus=true#step:11:151)
ModuleNotFoundError: No module named 'tensorflow.python.trackable'

**Comment by cery999** [comment]
> @cery999 I am the author of #11696. Thanks for your efforts in sending this PR. The reason why I did not fix this myself is that I have no idea whether this inconsistency should be regarded as a bug indeed, or an undefined behavior.
> 
> @masahi Would you please take a look at this PR and decide whether we should fix the inconsistency between `tanh` and `fast_tanh` on `nan` input, or leave it alone?

yes , I saw this issue , but find no fix, so I push it.

**Comment by cery999** [comment]
> 

I think this is turely a but, it the grad to be nan, it need to abandon this grad and do next iter, and grad scaler need to be updated , if the result will mistakenly be number, the grad scaler will never make sense. so we need to solve it

**Comment by masahi** [comment]
Please wait until https://github.com/apache/tvm/pull/12710 lands.

File: apps/android_camera/models/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-keras
+keras==2.9.0
 mxnet
 scipy
-tensorflow
\ No newline at end of file
+tensorflow
```
Comment (masahi) [inline apps/android_camera/models/requirements.txt:4]: Please drop this change

Comment (cery999) [reply apps/android_camera/models/requirements.txt:4]: I remove it.

**Comment by cery999** [comment]
> Please wait until #12710 lands.

fixed, @masahi

**Pull request merged**